python37Packages.black tests fail on Darwin

## Issue description

`python37Packages.black` tests fails on Darwin in sandbox mode:


```
running install tests
============================= test session starts ==============================
platform darwin -- Python 3.7.3, pytest-4.2.1, py-1.7.0, pluggy-0.8.1
rootdir: /private/var/folders/r2/zwnv7zsd4bd99mqkn4mhqxn800034l/T/nix-build-python3.7-black-19.3b0.drv-0/black-19.3b0, inifile:
collected 91 items / 3 deselected / 88 selected

tests/test_black.py ...FFFUnclosed client session
client_session: <aiohttp.client.ClientSession object at 0x1048d2860>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x102a83208>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x104fd1be0>
F.FFFUnclosed client session
client_session: <aiohttp.client.ClientSession object at 0x105103908>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x103536c88>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x1035d92b0>
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x1054311d0>
FFFF..................................... [ 59%]
....................................                                     [100%]

=================================== FAILURES ===================================
________________________ BlackTestCase.test_blackd_fast ________________________

self = <test_black.BlackTestCase testMethod=test_blackd_fast>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_fast(self) -> None:
        with open(os.devnull, "w") as dn, redirect_stderr(dn):
            app = blackd.make_app()
>           async with TestClient(TestServer(app)) as client:

tests/test_black.py:1508: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
________________ BlackTestCase.test_blackd_invalid_line_length _________________

self = <test_black.BlackTestCase testMethod=test_blackd_invalid_line_length>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_invalid_line_length(self) -> None:
        app = blackd.make_app()
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1531: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
_______________ BlackTestCase.test_blackd_invalid_python_variant _______________

self = <test_black.BlackTestCase testMethod=test_blackd_invalid_python_variant>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_invalid_python_variant(self) -> None:
        app = blackd.make_app()
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1439: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
____________________ BlackTestCase.test_blackd_line_length _____________________

self = <test_black.BlackTestCase testMethod=test_blackd_line_length>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_line_length(self) -> None:
        app = blackd.make_app()
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1521: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
________________________ BlackTestCase.test_blackd_pyi _________________________

self = <test_black.BlackTestCase testMethod=test_blackd_pyi>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_pyi(self) -> None:
        app = blackd.make_app()
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1463: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
___________________ BlackTestCase.test_blackd_python_variant ___________________

self = <test_black.BlackTestCase testMethod=test_blackd_python_variant>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_python_variant(self) -> None:
        app = blackd.make_app()
        code = (
            "def f(\n"
            "    and_has_a_bunch_of,\n"
            "    very_long_arguments_too,\n"
            "    and_lots_of_them_as_well_lol,\n"
            "    **and_very_long_keyword_arguments\n"
            "):\n"
            "    pass\n"
        )
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1484: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
______________ BlackTestCase.test_blackd_request_needs_formatting ______________

self = <test_black.BlackTestCase testMethod=test_blackd_request_needs_formatting>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_request_needs_formatting(self) -> None:
        app = blackd.make_app()
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1387: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
_________________ BlackTestCase.test_blackd_request_no_change __________________

self = <test_black.BlackTestCase testMethod=test_blackd_request_no_change>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_request_no_change(self) -> None:
        app = blackd.make_app()
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1397: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
________________ BlackTestCase.test_blackd_request_syntax_error ________________

self = <test_black.BlackTestCase testMethod=test_blackd_request_syntax_error>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_request_syntax_error(self) -> None:
        app = blackd.make_app()
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1406: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
_________________ BlackTestCase.test_blackd_supported_version __________________

self = <test_black.BlackTestCase testMethod=test_blackd_supported_version>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_supported_version(self) -> None:
        app = blackd.make_app()
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1429: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
________________ BlackTestCase.test_blackd_unsupported_version _________________

self = <test_black.BlackTestCase testMethod=test_blackd_unsupported_version>

    @unittest.skipUnless(has_blackd_deps, "blackd's dependencies are not installed")
    @async_test
    async def test_blackd_unsupported_version(self) -> None:
        app = blackd.make_app()
>       async with TestClient(TestServer(app)) as client:

tests/test_black.py:1419: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:392: in __aenter__
    await self.start_server()
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:253: in start_server
    await self._server.start_server(loop=self._loop)
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:105: in start_server
    _sock = get_port_socket(self.host, self.port)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

host = '127.0.0.1', port = 0

    def get_port_socket(host: str, port: int) -> socket.socket:
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>       s.bind((host, port))
E       PermissionError: [Errno 1] Operation not permitted

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:65: PermissionError
=============================== warnings summary ===============================
/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:194
  /nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:194: PytestWarning: cannot collect test class 'TestServer' because it has a __init__ constructor
    class TestServer(BaseTestServer):

/nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:226
  /nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/test_utils.py:226: PytestWarning: cannot collect test class 'TestClient' because it has a __init__ constructor
    class TestClient:

tests/test_black.py::BlackTestCase::test_blackd_fast
tests/test_black.py::BlackTestCase::test_blackd_invalid_line_length
tests/test_black.py::BlackTestCase::test_blackd_invalid_python_variant
tests/test_black.py::BlackTestCase::test_blackd_line_length
tests/test_black.py::BlackTestCase::test_blackd_main
tests/test_black.py::BlackTestCase::test_blackd_pyi
tests/test_black.py::BlackTestCase::test_blackd_python_variant
tests/test_black.py::BlackTestCase::test_blackd_request_needs_formatting
tests/test_black.py::BlackTestCase::test_blackd_request_no_change
tests/test_black.py::BlackTestCase::test_blackd_request_syntax_error
tests/test_black.py::BlackTestCase::test_blackd_supported_version
tests/test_black.py::BlackTestCase::test_blackd_unsupported_version
  /nix/store/rbjvv2hi1mdi4fx49ddf6jzpvq829gpq-python3.7-aiohttp-3.5.4/lib/python3.7/site-packages/aiohttp/web_urldispatcher.py:151: DeprecationWarning: Bare functions are deprecated, use async ones
    "use async ones", DeprecationWarning)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======= 11 failed, 77 passed, 3 deselected, 14 warnings in 12.68 seconds =======
builder for '/nix/store/9f7x4jwhx3g881nx5yik6f97ly8d87jb-python3.7-black-19.3b0.drv' failed with exit code 1
error: build of '/nix/store/9f7x4jwhx3g881nx5yik6f97ly8d87jb-python3.7-black-19.3b0.drv' failed
```

### Steps to reproduce

`nix-shell - p python37Packages.black`

## Technical details

 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.6.0, macOS 10.14.5`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(smaret): `"nixpkgs-19.03pre170546.0f43790d2ce"`
 - nixpkgs: `/nix/store/lgd8al5g6wd3hscir5bwskx1gzsvmirg-master.tar.gz`

--- a/pkgs/development/python-modules/black/default.nix
+++ b/pkgs/development/python-modules/black/default.nix
@@ -15,6 +15,10 @@ buildPythonPackage rec {
 
   checkInputs =  [ pytest glibcLocales ];
 
+  # Necessary for the tests to pass on Darwin with sandbox enabled.
+  # Black starts a local server and needs to bind a local address.
+  __darwinAllowLocalNetworking = true;
+
   # Don't know why these tests fails
   checkPhase = ''
     LC_ALL="en_US.UTF-8" pytest \

